### ISSUE TITLE
How to Optimize training performance for YOLOv8-OBB for Wrinkle Detection ?

### ISSUE THREAD

**Issue by Abhijeet241093** [ultralytics/ultralytics#19975]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello, thank you for the valuable resources. I have trained YOLOv8-OBB (n/m series) models using over 10,000 samples after preprocessing. Initially, the dataset included more than 2,000 samples of wrinkles, which is significantly higher than other defect types such as seal breakage, scratches, and rolling stickers. To address this imbalance, I applied the class balancing technique suggested by the Ultralytics team: [[YOLO Class Balancing](https://y-t-g.github.io/tutorials/yolo-class-balancing/)](https://y-t-g.github.io/tutorials/yolo-class-balancing/). I have also cleaned the entire dataset.  

However, I am still looking for ways to improve the training accuracy of the wrinkle class beyond **0.76**. Does anyone have any suggestions?  

I trained the model using the following approach:  

model.train(data="D:/Ginsuk_bottle/dataset/wrinkle/data.yaml", 
            epochs=200, 
            imgsz=1024, 
            conf=0.40,  
            lr0=0.0008,
            lrf=0.005,
            momentum=0.937,
            weight_decay=0.0005,
            warmup_epochs=3.0,
            warmup_momentum=0.8,
            warmup_bias_lr=0.1,
            multi_scale=True,
            mosaic=0.5,  # Reduce aggressive augmentations
            scale=0.2,   # Prevent extreme resizing
            shear=1.0,   # Minimal shearing to retain details
            flipud=0.2,  # Small probability of vertical flips
            cos_lr=True
    )  

However, when I attempted to train with `"yolov8m-obb.pt"` or `"yolov8x-obb.pt"`, my PC crashed due to resource exhaustion.  

I appreciate any suggestions to improve training accuracy. Thank you!  

Is yolov8n.pt more accurate than yolov8n-obb.pt ? 

### Additional

Thank you

**Comment by UltralyticsAssistant** [comment]
👋 Hi @Abhijeet241093, thank you for reaching out and for sharing detailed information about your use case 🚀! We appreciate your interest in optimizing YOLOv8-OBB for wrinkle detection.

We recommend checking out the [Ultralytics Docs](https://docs.ultralytics.com/) for guidance on training and optimization techniques. For additional tips on improving model performance, refer to our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). These resources may help refine your model further.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can investigate it effectively.

For resource exhaustion issues, consider:
- Using a smaller model like `yolov8n-obb.pt` or `yolov8s-obb.pt` if larger models like `yolov8m-obb.pt` or `yolov8x-obb.pt` are too demanding for your hardware.
- Leveraging cloud-based environments with GPUs, such as:
  - **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
  - **Google Cloud** Deep Learning VM ([GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/))
  - **Amazon** Deep Learning AMI ([AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/))
  - **Docker Image** ([Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/)) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

For further support, you can engage with the Ultralytics community:
- Join our Discord for real-time discussions: [Discord](https://discord.com/invite/ultralytics) 🎧
- Participate in in-depth discussions on [Discourse](https://community.ultralytics.com/)
- Share knowledge and ideas on our [Subreddit](https://reddit.com/r/Ultralytics)

Lastly, ensure you are running the latest version of the `ultralytics` package along with all requirements. You can upgrade with the following command:

```bash
pip install -U ultralytics
```

An Ultralytics engineer will review your query soon. Thank you for your patience 😊!

**Comment by glenn-jocher** [comment]
@Abhijeet241093 for optimizing YOLOv8-OBB training on wrinkle detection:

1. **Augmentation Balance**: Consider moderately increasing `mosaic` (0.75-1.0) while using `close_mosaic=10` to maintain detail retention while benefiting from augmentation. Our [OBB training guide](https://docs.ultralytics.com/tasks/obb/) shows optimal augmentation ranges.

2. **Class Focus**: Add `cls=2.0` in `model.train()` to emphasize wrinkle classification loss. For severe imbalances, focal loss can help - modify your loss function via custom code.

3. **Memory Efficiency**: When using larger models, add `batch=-1` (auto-batch) with `gradient_accumulation=2` to prevent crashes while maintaining effective batch size.

4. **Architecture**: `yolov8n-obb.pt` is specifically designed for OBB tasks and will outperform generic `yolov8n.pt` in oriented detection accuracy, despite similar base architecture.

For implementation details, see our [OBB training documentation](https://docs.ultralytics.com/tasks/obb/). Keep us posted on your progress! 🚀

**Comment by Abhijeet241093** [comment]
Yes, thank you for suggestion. Soon will share output result.

**Comment by Abhijeet241093** [comment]
![Image](https://github.com/user-attachments/assets/88fbc889-50c7-418a-b10b-d0564b6819df)

![Image](https://github.com/user-attachments/assets/64d943df-69c3-4fe7-be0e-242093899fe7)
